test/dns.server.test.ts

```diff
@@ -0,0 +1,187 @@
+import {
+  createHostedZone,
+  createRecord,
+  upsertRecord,
+  deleteRecord,
+  getChangeStatus,
+  RecordType,
+} from "../app/lib/dns.server";
+
+describe("DNS server lib function test", () => {
+  let hostedZoneId: string | undefined;
+
+  beforeAll(async () => {
+    hostedZoneId = await createHostedZone("starchart.com");
+  });
+
+  afterAll(() => resetHostedZone());
+
+  test("Hosted zone is created and hosted zone is returned", () => {
+    expect(hostedZoneId?.length).toBeGreaterThan(0);
+    expect(/^[A-Z0-9]+/.test(hostedZoneId!)).toBe(true);
+  });
+
+  test("createRecord() creates a record in existing hosted zone and returns changeId", async () => {
+    const changeId = await createRecord(
+      hostedZoneId!,
```

Comment:
You can drop the `!` here once you remove the `| undefined` on line 11.

Do the same throughout this file.